src/typesystem.rs

```diff
@@ -193,19 +193,28 @@ impl DataType {
         self.nature
     }
 
-    pub fn find_member(&self, member_name: &str) -> Option<&VariableIndexEntry> {
-        if let DataTypeInformation::Struct { members, .. } = self.get_type_information() {
-            members.iter().find(|member| member.get_name().eq_ignore_ascii_case(member_name))
-        } else {
-            None
+    pub fn find_member(&self, name: &str) -> Option<&VariableIndexEntry> {
+        match self.get_type_information() {
+            DataTypeInformation::Struct { members, .. }
+            | DataTypeInformation::Enum { variants: members, .. } => {
+                members.iter().find(|member| member.get_name().eq_ignore_ascii_case(name))
+            }
+            _ => None,
+        }
+    }
+
+    pub fn get_struct_members(&self) -> &[VariableIndexEntry] {
```

Comment:
Do we need both `find_member` and `find_struct_member`? Changing the single call to `find_struct_member` to `find_member` does not break any tests

Reply:
Found a case that shows it is needed after all:
The example in #1182 leads to a `recursive data structure` false-positive when calling `get_members` instead of `get_struct_members` in `RecursiveValidator::dfs`.
We should add a test for this.